collector/shards.go

```diff
@@ -133,21 +148,21 @@ func (s *Shards) Describe(ch chan<- *prometheus.Desc) {
 	for _, metric := range s.nodeShardMetrics {
 		ch <- metric.Desc
 	}
+	// Add new metric to Describe
+	ch <- indexShardAssignment
 }
 
 func (s *Shards) getAndParseURL(u *url.URL) ([]ShardResponse, error) {
 	res, err := s.client.Get(u.String())
 	if err != nil {
-		return nil, fmt.Errorf("failed to get from %s://%s:%s%s: %s",
+		return nil, fmt.Errorf("failed to get from %s://%s:%s%s: %w",
 			u.Scheme, u.Hostname(), u.Port(), u.Path, err)
 	}
-
 	defer func() {
-		err = res.Body.Close()
-		if err != nil {
+		if cerr := res.Body.Close(); cerr != nil {
```

Comment:
@dgur-outbrain if err you don't need to close the body. Actually idk what happens in case of error if close will fail or not